### PR TITLE
fix: duplicate X-Forwarded-Proto will be sent as string

### LINKED ISSUE #7215: bug: `X-Forwarded-Proto` header should not be set for upstream as a table
### Current Behavior

`X-Forwarded-Proto` header for upstream has been set as a table

### Expected Behavior

`X-Forwarded-Proto` header for upstream should be set as a string

### Error Logs

_No response_

### Steps to Reproduce

1. create a simple route:
```shell
curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -i -d '
{
    "uri": "/test",
    "upstream": {
        "type": "roundrobin",
        "nodes": {
            "127.0.0.1:8080": 1
        }
    }
}'
```

2. visit the route:
```shell
curl 127.0.0.1:9080/test -H "X-Forwarded-Proto: http" -H "X-Test: test"  -H "X-Forwarded-Proto: http" -H "X-Test: test"
```

3. the upstream will receive:
```shell
k: User-Agent, v: [curl/7.64.1]
k: X-Forwarded-For, v: [127.0.0.1]
k: X-Forwarded-Port, v: [9080]
k: X-Forwarded-Proto, v: [table: 0x1b70daa8]
k: X-Forwarded-Host, v: [127.0.0.1]
k: Accept, v: [*/*]
k: X-Test, v: [test test]
k: X-Real-Ip, v: [127.0.0.1]
```

see: 
https://github.com/apache/apisix/blob/master/apisix/init.lua#L230-L233

### Environment

- APISIX version (run `apisix version`): master branch
- Operating system (run `uname -a`): Darwin Kernel Version 19.6.0
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

### PR DESCRIPTION
**Pull request by soulbird**
### Description

duplicate request header: X-Forwarded-Proto will be sent as string instread of table address

Fixes #7215 

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/init.lua [hunk 1]
```diff
@@ -229,7 +230,11 @@ local function set_upstream_headers(api_ctx, picked_server)
 
     local hdr = core.request.header(api_ctx, "X-Forwarded-Proto")
     if hdr then
-        api_ctx.var.var_x_forwarded_proto = hdr
+        if type(hdr) == "table" then
+            api_ctx.var.var_x_forwarded_proto = core.table.concat(hdr, ", ")
```
Comment (spacewander) [inline apisix/init.lua:234]: It doesn't use `,` as the separator, see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto

Comment (soulbird) [reply apisix/init.lua:234]: So, X-Forwarded-Proto should be unique, not multiple?

File: t/node/x-forwarded.t [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+
+
+
+=== TEST 2: duplicate request header: X-Forwarded-Proto
```
Comment (spacewander) [inline t/node/x-forwarded.t:62]: Please add this test to https://github.com/apache/apisix/blob/master/t/plugin/proxy-rewrite2.t

File: t/node/x-forwarded.t [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+--- request
+GET /echo
+--- more_headers
+X-Forwarded-Proto: http
+X-Forwarded-Proto: grpc
+--- error_code: 200
```
Comment (spacewander) [inline t/node/x-forwarded.t:68]: The 200 status code is checked by default

File: t/node/x-forwarded.t (hunk 2 above)
Comment (soulbird) [reply t/node/x-forwarded.t:62]: This doesn't seem to have anything to do with the proxy-rewrite plugin.

**Review by spacewander — APPROVED** [review]

**Review by nic-chen — APPROVED** [review]

**Review by bzp2010 — APPROVED** [review]

**Pull request merged**